src/auth/src/credentials.rs

```diff
@@ -201,6 +201,41 @@ pub(crate) mod dynamic {
     }
 }
 
+/// Represents the type of credential as specified in the credentials file.
+#[derive(Debug, PartialEq)]
+pub enum CredentialType {
```

Comment:
Hm, this is public, but we are not exposing any way to go from a `Credential` -> `CredentialType` anywhere. Can we make it `pub(crate)` for now? Why should it be `pub`?

FYI: I think our plan in the long run is to have metadata like `CredentialType` available, at least in the error messages, so users can debug auth configuration errors.